_posts/dotnet-development-on-apple-silicon.md

```diff
@@ -0,0 +1,75 @@
+header_image: /img/posts/dotnet-development-on-apple-silicon/header.jpg
+tags:
+- development
+---
+
+I've used a Dell Precision Tower 5810 workstation (8 core Xeon, 32GB of RAM, and 1TB SSD) for the last seven years and Windows as my primary OS for most of my career. I had an experience with macOS for a couple of years around 2010.
+
+I recently (April 2022) bought a MacStudio (M1 Ultra, 64GB of RAM, 1TB SSD).
+
+This post intends to be a live-blogging thing, sharing my experience as a long-time Windows user and, more importantly, a .NET developer. I started with .NET development in 1999 with one of the first alpha releases.
+
+<blockquote class="twitter-tweet"><p lang="en" dir="ltr">Let’s get started 😬 <a href="https://t.co/K55bOMQojn">pic.twitter.com/K55bOMQojn</a></p>&mdash; Mauro Servienti 🐙 (@mauroservienti) <a href="https://twitter.com/mauroservienti/status/1511755988453498881?ref_src=twsrc%5Etfw">April 6, 2022</a></blockquote> <script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+<!-- let's get started -->
+
+Before getting into the nitty-gritty details of .NET development using an M1 chip and macOS, let me give you an overview of the hardware. I'm famous among my friends and colleagues for my diplomacy.
+
+The MacStudio built-in audio quality is shit. I had never heard something so utterly inadequate. I should have expected it. Apple, during the presentation, never mentioned the audio using their usual style, "The best XYZ feature of all times." And so, there we go. Even for regular Zoom calls, you need external speakers or headphones.
+
+Having a lot of thunderbolt ports, six on my model (four in the back and two on the front), is a huge bonus. Plugging in devices has never been so easy. Except if you want to plugin not-officially supported USB-C screens.
+
+I have two Dell 32" 4K screens (U3219Q) claiming to support DisplayPort over the USB-C connection. But things are not so easy. I'm not sure why, but it sounds like external screens and projectors are more or less always a source of pain with Apple machines.
+
+The problem manifests itself at boot time. The screen is in standby mode, the machine starts, the screen wakes up and starts the USB-C handshaking process, but that happens too early, and the response from the Mac times out. The screen never retries and never connects. The "easy" workaround is disconnecting and reconnecting the USB-C cable, and the screen immediately starts working. However, if you're using the screen as a USB hub, that causes a mess with USB devices connected to the screen. Dell forums are full of people complaining about this issue, so it's not something new to the MacStudio, and it's not easy to understand who to blame. Anyway, the definitive solution is to use a USB-C to DisplayPort cable. No more problems. The downside is that to use the screens as a USB hub, I had to connect them using the USB-A cable, which is fine because the MacStudio comes with two USB-A ports.

```

Comment:
```suggestion
The problem manifests itself at boot time. The screen is in standby mode, the machine starts, the screen wakes up and starts the USB-C handshaking process. But that happens too early, and the response from the Mac times out. The screen never retries and never connects. The "easy" workaround is disconnecting and reconnecting the USB-C cable, and the screen immediately starts working. However, if you're using the screen as a USB hub, that causes a mess with USB devices connected to the screen. Dell forums are full of people complaining about this issue, so it's not something new to the MacStudio, and it's not easy to understand who to blame. Anyway, the definitive solution is to use a USB-C to DisplayPort cable. No more problems. The downside is that to use the screens as a USB hub, I had to connect them using the USB-A cable, which is fine because the MacStudio comes with two USB-A ports.

```